cmd/libvirt-actuator/README.md

```diff
@@ -2,6 +2,180 @@
 
 The command allows to directly interact with the libvirt actuator.
 
+## Prerequisities
+
+1. Download qcow2 Fedora 28 cloud image from https://alt.fedoraproject.org/cloud/
+1. Generate your cloud-init file under `cloudinitdata`, e.g.:
+   ```
+   #cloud-config
+
+   # Hostname management
+   preserve_hostname: False
+   hostname: $1
+   fqdn: $1.example.local
+
+   runcmd:
+     # all user data (written in bash) are expected under /sys/firmware/qemu_fw_cfg/by_name/opt/com.coreos/config/raw
+     - sudo bash /sys/firmware/qemu_fw_cfg/by_name/opt/com.coreos/config/raw
+     # Remove cloud-init when finished with it
+     - [ yum, -y, remove, cloud-init ]
+
+   # Configure where output will go
+   output:
+     all: ">> /var/log/cloud-init.log"
+
+   # username: fedora, password: fedora
+   # for debugging purposes
+   password: fedora
+   chpasswd: { expire: False }
+   ssh_pwauth: True
+
+   # configure interaction with ssh server
+   ssh_svcname: ssh
+   ssh_deletekeys: True
+   ssh_genkeytypes: ['rsa', 'ecdsa']
+   ```
+1. Generate iso with the cloud init file:
+  ```sh
+  genisoimage -output cloud-init.iso -volid cidata -joliet -r cloudinitdata
+  ```
+
+1. Provision instance in `packet.net`:
+  ```sh
+  ./hack/packet-provision.sh install
+  ```
+  and get the IP address (assuming it is `147.75.96.139`).
+
+1. Update ``~/.ssh/config`` to contain:
+  ```
+  Host libvirtactuator 147.75.96.139
+  Hostname 147.75.96.139
+  ControlPersist 10m
+  ControlMaster auto
+  ControlPath /tmp/%r@%h:%p
+  User root
+  StrictHostKeyChecking no
+  PasswordAuthentication no
+  UserKnownHostsFile ~/.ssh/aws_known_hosts
+  IdentityFile /tmp/packet_id_rsa
+  IdentitiesOnly yes
+  Compression yes
+  ```
+1. Run init script to create additional resources in the instance:
+  ```sh
+  #!/bin/sh
+
+  cat <<EOF |
+  [centos]
+  name=centos
+  baseurl=http://mirror.centos.org/centos/7/virt/x86_64/kvm-common/
+  gpgcheck=0
+  enabled=1
+  EOF
+  ssh libvirtactuator 'cat > /etc/yum.repos.d/centos.repo'
+
+  ssh libvirtactuator 'yum update qemu-kvm -y'
+
+  export VIRSH="virsh -c qemu+ssh://root@147.75.96.139/system"
+
+  ${VIRSH} pool-define /dev/stdin <<EOF
+  <pool type='dir'>
+    <name>default</name>
+    <target>
+      <path>/var/lib/libvirt/images</path>
+    </target>
+  </pool>
+  EOF
+
+  ${VIRSH} pool-start default
+  ${VIRSH} pool-autostart default
+  ```
+
+1. Create `coreos_base` volume by running the following terraform content:
+  ```
+  variable "libvirturi" {
+    type    = "string"
+  }
+
+  provider "libvirt" {
+    uri = "${var.libvirturi}"
+  }
+
+  resource "libvirt_volume" "coreos_base" {
+    name   = "coreos_base"
+    source = "https://download.fedoraproject.org/pub/fedora/linux/releases/28/Cloud/x86_64/images/Fedora-Cloud-Base-28-1.1.x86_64.qcow2"
+  }
+  ```
+  with `terraform apply -var libvirturi=qemu+ssh://root@147.75.96.139/system -auto-approve`
+
+1. Provider user data (under `user-data.sh` file) copied to the packet instances. E.g.:
```

Comment:
Used data copy pasted from https://github.com/openshift/cluster-api-provider-aws/blob/master/examples/master-user-data.sh plus installing the `docker`